partiql-parser/src/scanner.rs

```diff
@@ -9,15 +9,19 @@
 
 use crate::peg::{PairExt, PairsExt, PartiQLParser, Rule};
 use crate::prelude::*;
-use crate::result::syntax_error;
 use pest::iterators::Pair;
 use pest::Parser;
+use std::borrow::Cow;
 
 /// The parsed content associated with a [`Token`] that has been scanned.
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub enum Content<'val> {
-    /// A PartiQL keyword.  Contains the slice for
-    Keyword(&'val str),
+    /// A PartiQL keyword.  Contains the slice for the keyword case folded to upper case.
+    Keyword(Cow<'val, str>),
+    /// An identifier.  Contains the slice for the text of the identifier.
+    Identifier(Cow<'val, str>),
+    /// A string literal.  Contains the slice for the content of the literal.
+    String(Cow<'val, str>),
```

Comment:
A `Cow` can be either an *owned* value (`String`) or a borrowed reference.  This gives us a bit more flexibility to use the raw input when we can, but construct a heap allocated object if we need to manipulate it.

https://doc.rust-lang.org/std/borrow/enum.Cow.html